Blog.API/Blog.Infrastructure/Services/CategoryService.cs

```diff
@@ -1,36 +1,106 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Threading.Tasks;
+﻿using Blog.Core.Contracts.Controllers.Categories;
 using Blog.Core.Entities;
+using Blog.Infrastructure.Abstract.Interfaces;
 using Blog.Infrastructure.Services.Interfaces;
+using FluentValidation;
+using MapsterMapper;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 
 namespace Blog.Infrastructure.Services;
 
-//TODO:Implement methods
 public sealed class CategoryService : ICategoryService
 {
-    public IEnumerable<Category> GetCategories()
+    private readonly IUnitOfWork<Category> _unitOfWork;
+    private readonly ILogger _logger;
+    private readonly IMapper _mapper;
+
+    public CategoryService(IUnitOfWork<Category> unitOfWork,
+        ILogger<CategoryService> logger,
+        IMapper mapper)
     {
-        throw new NotImplementedException();
+        _unitOfWork = unitOfWork;
+        _logger = logger;
+        _mapper = mapper;
     }
 
-    public Task<Category?> GetCategoryById(Guid id)
+    public async Task<IEnumerable<CategoryResponse>> GetCategories()
     {
-        throw new NotImplementedException();
+        var categories = await _unitOfWork.GenericRepository.Set
+            .Include(c => c.Posts)
+            .ToListAsync();
+
+        foreach (var category in categories)
+        {
+            if (category.Posts is not null)
+            {
+                foreach (Post post in category.Posts)
+                {
+                    post.Category = null;
+                }
+            }
+        }
+        return _mapper.Map<IEnumerable<CategoryResponse>>(categories);
     }
 
-    public Task<Category> CreateCategory(Category createCategory)
+    public async Task<CategoryResponse?> GetCategoryById(Guid id)
     {
-        throw new NotImplementedException();
+        Category? category = await _unitOfWork.GenericRepository.Set
+            .Where(c => c.CategoryId == id)
+            .Include(c => c.Posts)
+            .FirstOrDefaultAsync();
+
+        if (category is not null)
+        {
+            if (category.Posts is not null)
+            {
+                foreach (Post post in category.Posts)
+                {
+                    post.Category = null;
+                }
+            }
+            return _mapper.Map<CategoryResponse>(category);
+        }
+
+        _logger.LogError("Category object with id {id} doesn't exist", id);
```

Comment:
You need to change the name of the variable in the template. This value will be used in logs later. We will use the name of the variable to search for them. It should be something like this(it just an example, try to find better name):
```suggestion
        _logger.LogError("Category object with id {GetCategoryByIdId} doesn't exist", id);
```